contracts/controller/src/prices.rs

```diff
@@ -24,6 +22,20 @@ impl Contract {
     pub fn get_price(&self, dtoken_id: AccountId) -> Option<Price> {
         self.prices.get(&dtoken_id)
     }
+
+    pub fn get_price_by_ticker(&self, ticker: String) -> Price {
+        let mut result = vec![];
+
+        for (_, price) in self
+            .prices
+            .iter()
+            .filter(|(_, market)| market.ticker_id == ticker)
+        {
+            result.push(price);
+        }
+
+        result.pop().unwrap()
+    }
```

Comment:
Just FYI: When you are faced with a task to find smth and return, don't use buffer struct for this. You can use the find method.

    pub fn get_price_by_ticker(&self, ticker_id: String) -> Option<Price> {
        self.prices
            .values()
            .iter()
            .find(|(_, price)| price.ticker_id == ticker_id)
    }